src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -788,5 +1003,89 @@ private void Write(ProfiledLocation e)
             Write(e.ExclusiveTime);
             Write(e.InclusiveTime);
         }
+
+        internal struct HashKey : IEquatable<HashKey>
+        {
+            private ulong value;
```

Comment:
```suggestion
        internal readonly struct HashKey : IEquatable<HashKey>
        {
            private readonly ulong value;
```

Reply:
I was actually thinking of deleting the HashKey struct and just using ulong directly... still on the fence. Thoughts?